internal/super/files.go

```diff
@@ -0,0 +1,231 @@
+)
+
+type accountChange struct {
+	status int
+	name   string
+}
+
+type contractChange struct {
+	status  int
+	path    string
+	account string
+}
+
+func newProjectFiles(projectPath string) *projectFiles {
+	return &projectFiles{
+		cadencePath: path.Join(projectPath, cadenceDir),
+		watcher:     watcher.New(),
+	}
+}
+
+type projectFiles struct {
+	cadencePath string
+	watcher     *watcher.Watcher
+}
+
+// exist checks if current directory contains all project files required.
+func (f *projectFiles) exist() error {
+	if _, err := os.Stat(f.cadencePath); errors.Is(err, os.ErrNotExist) {
+		return fmt.Errorf("required cadence folder does not exist")
+	}
+	if _, err := os.Stat(config.DefaultPath); errors.Is(err, os.ErrNotExist) {
+		return fmt.Errorf("required project configuration ('flow.json') does not exist")
+	}
+
+	return nil
+}
+
+// contracts returns a list of contracts in project.
+func (f *projectFiles) contracts() ([]string, error) {
+	return f.getFilePaths(contractDir)
+}
+
+// contracts returns a map of deployments in project, mapping account name to all contracts saved to that account.
+func (f *projectFiles) deployments() (map[string][]string, error) {
+	deployments := make(map[string][]string)
+
+	contracts, err := f.contracts()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get contracts in deployment")
+	}
+
+	for _, file := range contracts {
+		accName, _ := accountFromPath(file)
+		deployments[accName] = append(deployments[accName], file)
+	}
+
+	return deployments, nil
+}
+
+// contracts returns a list of scripts in project.
+func (f *projectFiles) scripts() ([]string, error) {
+	return f.getFilePaths(scriptDir)
+}
+
+// contracts returns a list of transactions in project.
+func (f *projectFiles) transactions() ([]string, error) {
+	return f.getFilePaths(transactionDir)
+}
+
+// watch for file changes in the contract folder and signal any changes through channel.
+//
+// This function returns two channels, accountChange which reports any changes on the accounts folders and
+// contractChange which reports any changes to the contract files.
+func (f *projectFiles) watch() (<-chan accountChange, <-chan contractChange, error) {
+	err := f.watcher.AddRecursive(path.Join(f.cadencePath, contractDir))
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "add recursive files failed")
+	}
+
+	go func() {
+		err = f.watcher.Start(500 * time.Millisecond)
+		if err != nil {
+			panic(err)
+		}
+	}()
+
+	accounts := make(chan accountChange)
+	contracts := make(chan contractChange)
+
+	go func() {
+		status := map[watcher.Op]int{
+			watcher.Create: created,
+			watcher.Remove: removed,
+			watcher.Write:  changed,
+		}
+
+		for {
+			select {
+			case event := <-f.watcher.Event:
+				rel, err := f.relProjectPath(event.Path)
+				if err != nil { // skip if failed
+					continue
+				}
+
+				name, containsAccount := accountFromPath(rel)
+				if event.IsDir() && containsAccount {
+					// todo handle rename and move
```

Comment:
Separate PR for this later?

Reply:
yeah, this is a bit edge case I feel, we can cover later